patroni/utils.py

```diff
@@ -1064,6 +1064,60 @@ def data_directory_is_empty(data_dir: str) -> bool:
     return all(os.name != 'nt' and (n.startswith('.') or n == 'lost+found') for n in os.listdir(data_dir))
 
 
+def apply_keepalive_limit(option: str, value: int) -> int:
+    """
+    Adjusts the keepalive option value to ensure it falls within the acceptable range for the current operating system.
+
+    :param option: The TCP keepalive option name.
+        Possible values are "TCP_USER_TIMEOUT", "TCP_KEEPIDLE", "TCP_KEEPINTVL", and "TCP_KEEPCNT".
+    :param value: The desired value for the keepalive option.
+
+    Returns:
+    :returns: The adjusted value that falls within the acceptable range for the specified keepalive option.
+    """
+    min_possible_value = 1
+    max_possible_value = None
+    if sys.platform.startswith('linux'):
+        if option == "TCP_USER_TIMEOUT":
+            max_possible_value = 2147483647
+        if option == "TCP_KEEPIDLE":
+            max_possible_value = 32767
+        if option == "TCP_KEEPINTVL":
+            max_possible_value = 32767
+        if option == "TCP_KEEPCNT":
+            max_possible_value = 127
+
+    if sys.platform.startswith('darwin'):
+        if option == "TCP_KEEPIDLE":
+            max_possible_value = 4294967
+        if option == "TCP_KEEPINTVL":
+            max_possible_value = 4294967
+        if option == "TCP_KEEPCNT":
+            max_possible_value = 2147483647
+
+    if sys.platform.startswith('win32'):
+        if option == "TCP_KEEPIDLE":
+            max_possible_value = 2147483647
+        if option == "TCP_KEEPINTVL":
+            max_possible_value = 2147483647
+        if option == "TCP_KEEPCNT":
+            max_possible_value = 255
```

Comment:
That's unknown territory, and has to be verified how it works.
So far only enable_keepalive() was handling the Windows case by using `socket.ioctl`.

Reply:
Ok, sure. For now i'll remove this section.

Reply:
I mean, some people for sure running Patroni on Windows in production, and this change might affect them.
Maybe positive, but I don't know, it must be tested.

Reply:
I understand. However, I think we can proceed with this PR by temporarily excluding Windows and deferring the tests for now. Testing this scenario might be a bit complex, and there is currently a bug in Linux.